src/gui/qgsattributeform.cpp

```diff
@@ -1770,6 +1770,8 @@ void QgsAttributeForm::init()
             {
               tabWidget = new QgsTabWidget();
               layout->addWidget( tabWidget, row, column, 1, 2 );
+              // Use row stretch to allocate more vertical space for tab widget.
```

Comment:
I don't think this is quite the right fix -- elsewhere in this file we use addSpacer = false to avoid adding the unwanted space at the bottom of the form. Does that approach work here?